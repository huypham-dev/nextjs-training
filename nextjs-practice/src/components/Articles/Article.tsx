import React from "react";
import Link from "next/link";
import Image from "next/image";

const Article = ({ article }) => {
  return (
    <article className="mb-12">
      <div className="float-left">
        <Link href="#">
          <a>
            <Image src={article.image} width={150} height={150} />
          </a>
        </Link>
      </div>
      <div>
        <h2 className="mb-4">
          <a
            href="#"
            className="text-black text-xl md:text-2xl no-underline hover:underline"
          >
            {article.title}
          </a>
        </h2>

        <div className="mb-4 text-sm text-gray-700">
          by{" "}
          <a href="#" className="text-gray-700">
            {article.author}
          </a>{" "}
          {article.createdAt}
          <span className="font-bold mx-1"> | </span>
          <a href="#" className="text-gray-700">
            {article.category}
          </a>
        </div>

        <p className="text-gray-700 leading-normal">
          {article.content}
        </p>
      </div>
    </article>
  );
};

export default Article;
