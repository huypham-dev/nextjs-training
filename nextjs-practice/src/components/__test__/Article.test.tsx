import { mount, shallow } from "enzyme";
import Article from "../Articles/Article";

const article = {
  id: "1",
  createdAt: "2021-03-01T12:59:16.855Z",
  title: "Lorem ipsum dolor sit amet, consectetur adipisicing elit",
  author: "author 1",
  image:
    "https://genk.mediacdn.vn/139269124445442048/2021/3/2/photo-1-16146533712881100681556.jpg",
  category: "internet",
  content:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.1",
  slug: "slug 1",
  view: 90,
};

describe("Article", () => {
  it("Snapshot", () => {
    const component = shallow(<Article article={article} />);

    expect(component).toMatchSnapshot();
  });

  it("Render component correctly", () => {
    const component = mount(<Article article={article} />);
    expect(
      component.containsMatchingElement(
        <a>Lorem ipsum dolor sit amet, consectetur adipisicing elit</a>
      )
    ).toBeTruthy();
    expect(component.containsMatchingElement(<div>internet</div>)).toBeTruthy();
    expect(
      component.containsMatchingElement(
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
          eiusmod tempor incididunt ut labore ...
        </p>
      )
    ).toBeTruthy();
  });
});
