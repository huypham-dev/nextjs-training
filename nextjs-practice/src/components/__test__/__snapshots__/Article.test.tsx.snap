// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Snapshot 1`] = `
<article
  className="max-w-md mx-auto rounded-md shadow-md overflow-hidden md:max-w-2xl mb-12"
>
  <div
    className="md:flex"
  >
    <Memo(LinkBase)
      className="flex flex-shrink-0 justify-center md:items-center"
      href="/post/1"
    >
      <Image
        alt="article image"
        height={150}
        src="https://genk.mediacdn.vn/139269124445442048/2021/3/2/photo-1-16146533712881100681556.jpg"
        width={250}
      />
    </Memo(LinkBase)>
    <div
      className="p-8"
    >
      <div
        className="uppercase tracking-wide text-sm text-indigo-500 font-semibold"
      >
        internet
      </div>
      <Memo(LinkBase)
        className="block mt-1 text-lg leading-tight font-medium text-black hover:underline"
        href="/post/1"
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit
      </Memo(LinkBase)>
      <Memo(Typography)
        className="mt-2 text-gray-500"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore 
        ...
      </Memo(Typography)>
    </div>
  </div>
</article>
`;
